packages/@lightningjs/ui-components/src/components/NavigationManager/NavigationManager.d.ts

```diff
@@ -42,35 +42,152 @@ export type NavigationManagerStyles = {
+
+    /**
+     * If true, the component will never scroll, unless `alwaysScroll` is set to true.
+     * If false, the component will apply normal scrolling logic.
+     */
+    neverScroll?: boolean;
+
+    /**
+     * Item index at which scrolling begins, provided the sum of item widths is greater than the width of the Row component.
+     */
+    scrollIndex?: number;
+
+    /**
+     * Automatically sets the width of the row to be the same as the Items container inside (this will prevent scrolling).
+     */
+    autoResizeWidth?: boolean;
+
+    /**
+     * Automatically sets the height of the row to be the same as the Items container inside.
+     */
+    autoResizeHeight?: boolean;
+
+    /**
+     * If set, this is used to calculate the initial number of items to display on the first render (`lazyUpCount` + `lazyLoadUpCountBuffer`).
+     * The remaining items are stored as lazy items.
+     * Each time the user navigates further, the next lazy item will be added.
+     */
+    lazyUpCount?: number;
+
+    /**
+     * If `lazyUpCount` is set, this is used to calculate the initial number of items to display on the first render (`lazyUpCount` + `lazyLoadUpCountBuffer`).
+     * The remaining items are stored as lazy items.
+     * Each time the user navigates further, the next lazy item will be added.
+     */
+    lazyUpCountBuffer?: number;
+  }
+}
+
+declare class NavigationManager<
+  TemplateSpec extends NavigationManager.TemplateSpec = NavigationManager.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
+> extends FocusManager<TemplateSpec, TypeConfig> {
+  // Properties
+  /**
+   * Determines whether the component will stop scrolling as it nears the end to prevent white space.
+   * If true, the component will scroll infinitely.
+   */
   alwaysScroll?: boolean;
+
+  /**
+   * If true, the component will never scroll, unless alwaysScroll is set to true.
+   * If false, the component will apply normal scrolling logic.
+   */
   neverScroll?: boolean;
+
+  /**
+   * Item index at which scrolling begins, provided the sum of item widths is greater than the width of the Row component.
+   */
   scrollIndex?: number;
+
+  /**
+   * Automatically sets the width of the row to be the same as the Items container inside (this will prevent scrolling).
+   */
   autoResizeWidth?: boolean;
+
+  /**
+   * Automatically sets the height of the row to be the same as the Items container inside.
+   */
   autoResizeHeight?: boolean;
 
+  /**
+   * If set, this is used to calculate the initial number of items to display on the first render (`lazyUpCount` + `lazyLoadUpCountBuffer`).
+   * The remaining items are stored as lazy items.
+   * Each time the user navigates further, the next lazy item will be added.
+   */
+  lazyUpCount?: number;
+
+  /**
+   * If `lazyUpCount` is set, this is used to calculate the initial number of items to display on the first render (`lazyUpCount` + `lazyLoadUpCountBuffer`).
```

Comment:
Took this property description from the `NavigationManager` documentation but it appears to be almost the same as the `lazyUpCount` property description. Is this expected?

Reply:
I feel like we could update this to provide better documentation. Something explaining how to use `lazyUpCount` and the buffer in tandem to control how items are loaded

Reply:
Will do! Going to reach out to @THoj13 for the best way to describe these properties